GEMCosmicMuonSim Part

slowmoyang opened the pull request:
I referred to cmssw/Validation/MuonGEMRecHits/src/GEMRecHitsValidation.cc

1. Passed / Total SimHit  per VFAT
simhit / rechit matching condition is to check whether a fired strip by simhit is in a rec cluster.
Clusters may overlap. In that case, I select first cluster.
The local position of simhit is used to get fired strip by simhit. 
Fired strips by simhit start from 0 but Fired strips by rechit start from 1.
Then I draw the efficiency plot using a genDrawSim python script in the test directory.

2. Residual and Pull (on Local Coordinates)
Residual u = < u of RecHit > - < u of SimHit > for u =x, y, phi. (phi 
Pull U = <Residual u> /  <local position error of RecHit> for u =x, y
To get local phi of hit, I use strip number.

3. Occupancy Per VFAT
I just fill histograms over the all RecHit..
Is it right?

slowmoyang committed:
Merged GEMCosmicMuon-10_1_0 from repository jshlee with cms-merge-topic

slowmoyang committed:
Add sim and utility scripts

slowmoyang committed:
Add GEMCosmicStandSim scripts and utility scripts

slowmoyang committed:
Add checking matching between recHits and simHits

slowmoyang committed:
Merge branch 'GEMCosmicMuon-10_1_0' of https://github.com/slowmoyang/cmssw into GEMCosmicMuon-10_1_0

slowmoyang committed:
Merged GEMCosmicMuon-10_1_0 from repository jshlee with cms-merge-topic

slowmoyang committed:
Update scripts for GEMCosmicMuonSim

slowmoyang committed:
Update files to fix conflicts

slowmoyang committed:
Remove git conflict traces

slowmoyang committed:
Merged GEMCosmicMuon-10_1_0 from repository jshlee with cms-merge-topic

slowmoyang committed:
Update

slowmoyang committed:
Merge branch 'GEMCosmicMuon-10_1_0' of https://github.com/slowmoyang/cmssw into GEMCosmicMuon-10_1_0

slowmoyang committed:
TODO: analysing mismatching case

slowmoyang committed:
Fix typo in argument of VFAT total/passed monitor element

slowmoyang committed:
ApMake utility scripts available on kisti and gate

slowmoyang committed:
fix typo

slowmoyang committed:
use local position of SimHit instead of entry point to get fired strip.

slowmoyang committed:
Add lines for occupancy

slowmoyang committed:
Change filename of 'gemDraw' to 'gemDrawSim'

slowmoyang committed:
fix gemDrawSim

jshlee commented on Validation/GEMCosmicMuonStand/test/gemClearReco:
```diff
@@ -0,0 +1,11 @@
+#!/bin/sh
```
dont add this file

jshlee commented on Validation/GEMCosmicMuonStand/test/gemClearSim:
```diff
@@ -0,0 +1,3 @@
+#!/bin/sh
```
dont add this file

jshlee commented on Validation/GEMCosmicMuonStand/test/gemMakeListOfOutDigi:
```diff
@@ -0,0 +1,29 @@
+#!/usr/bin/env python
```
try to have generic scripts

jshlee commented:
since this will be used by other people, try not to have kisti only scripts

slowmoyang committed:
1. GEMCosmicStandSim
Add monitor elements for residual and pull per VFAT

2. gemDrawSim
Change the structure of a plotting script.
One output histogram is created by one Painter instance.
There are two types of Painter class:
    HistPainter - Painter for the histogram
    EffPainter for the efficiency
And both of them inherit from the BasePainter.
